src/lib.rs

```diff
@@ -108,25 +111,40 @@ impl AsRef<u8> for u5 {
     fn as_ref(&self) -> &u8 { &self.0 }
 }
 
-/// Interface to write `u5`s into a sink
-pub trait WriteBase32 {
+/// Interface to write `u(2*n)`s into a sink
+pub trait WriteBaseN<T>
```

Comment:
In be4af63082fdbe52c7de6cb5186c2f51be6a7d0b:

I find this rename very confusing. It looks like it's generic over the number base, but the `write` implementation still calls `write_u5` ... which now no longer has anything to do with u5s.

I think we should leave this trait alone and make `WriteBase256` completely independent (and probably it does not need the single `write_u5` method since Rust has decent first-class support for byteslices).

Reply:
Oops not renaming `write_u5` was just an oversight.

I kept it one trait to I can have the polymorphic `Vec` and `ArrayVec` implementations, but I suppose I could just copy paste them for each trait.

Reply:
I would suggest copy/pasting because I think `u8` serialization is "just dumping bytes" and `u5` serialization as a nontrivial conversion, so they feel like distinct things to me even though the code is pretty-much the same.

Reply:
OK